autotest/gdrivers/isis.py

```diff
@@ -2043,3 +2043,142 @@ def test_isis3_oblique_cylindrical_write():
     ds = None
 
     gdal.GetDriverByName("ISIS3").Delete("/vsimem/isis_tmp.lbl")
+
+
+def test_isis_read_data(tmp_vsimem):
+    gdal.FileFromMemBuffer(
+        tmp_vsimem / "in.lbl",
+        """Object = IsisCube
+  Object = Core
+    StartByte = 1
+    Format = BandSequential
+    Group = Dimensions
+      Samples = 1
+      Lines   = 1
+      Bands   = 1
+    End_Group
+    Group = Pixels
+      Type       = UnsignedByte
+      ByteOrder  = Lsb
+      Base       = 0.0
+      Multiplier = 1.0
+    End_Group
+  End_Object
+End_Object
+
+Object = Table
+  Name = FirstTable
+  StartByte = 1001
+  Bytes = 10
+End_Object
+
+Object = Table
+  Name = SecondTable
+  StartByte = 1
+  Bytes = 5
+  ^Table = in.bin
+End_Object
+
+Object = Table
+  Name = ThirdTable
+  StartByte = 1
+  Bytes = 5
+  ^Table = not_existing.bin
+End_Object
+
+End
+""",
+    )
+
+    gdal.FileFromMemBuffer(tmp_vsimem / "in.bin", "b\x01\x02\x03\x04\x05")
+
+    f = gdal.VSIFOpenL(tmp_vsimem / "in.lbl", "rb+")
+    assert f
+    gdal.VSIFTruncateL(f, 1000)
+    gdal.VSIFSeekL(f, 1000, 0)
+    gdal.VSIFWriteL("0123456789", 10, 1, f)
+    gdal.VSIFCloseL(f)
+
+    with gdal.Open(tmp_vsimem / "in.lbl") as ds:
+        got = json.loads(ds.GetMetadata("json:ISIS3")[0])
+    assert "_data" in got["Table_FirstTable"]
+    del got["_filename"]
+    expected = {
+        "IsisCube": {
+            "Core": {
+                "Dimensions": {
+                    "Bands": 1,
+                    "Lines": 1,
+                    "Samples": 1,
+                    "_type": "group",
+                },
+                "Format": "BandSequential",
+                "Pixels": {
+                    "Base": 0.0,
+                    "ByteOrder": "Lsb",
+                    "Multiplier": 1.0,
+                    "Type": "UnsignedByte",
+                    "_type": "group",
+                },
+                "StartByte": 1,
+                "_type": "object",
+            },
+            "_type": "object",
+        },
+        "Table_FirstTable": {
+            "Bytes": 10,
+            "Name": "FirstTable",
+            "StartByte": 1001,
+            "_container_name": "Table",
+            "_data": {
+                "ASCII": "0123456789",
+            },
```

Comment:
Seems different from the implementation here: https://github.com/DOI-USGS/ISIS3/discussions/5698, we can change how ISIS ingests it as GDAL changes are still in dev, but im not sure 100% when table data will be anything other than either an external table or hex

I imagine how fields are converted to JSON is not different?

Reply:
yes, I tried a slightly different approach. I chose "_data" as a member name, with the leading underscore being the GDAL convention for things that are not 1:1 equivalent of content from the PVL label (like "_type", "_container_name").
Regarding exposing either a ASCII or HEX sub-member, I thought that exposing as ASCII, when the content is, would be slightly more user friendly, but I can understand this also complicates the decoding logic.
If you prefer to have just ``"_data": "HEX_CONTENT_HERE"``, I'd fine changing to that. Just tell your preference.

Reply:
I think the key change is cleaner, but I would prefer keeping it to hex.